packages/headless/src/controllers/product-listing/result-list/headless-product-listing-interactive-result.ts

```diff
@@ -0,0 +1,58 @@
+import {Result} from '../../../api/search/search/result';
+import {pushRecentResult} from '../../../features/recent-results/recent-results-actions';
+import {logDocumentOpen} from '../../../features/result/result-analytics-actions';
+import {ProductListingEngine} from '../../../product-listing.index';
+import {
+  buildInteractiveResultCore,
+  InteractiveResultCore,
+  InteractiveResultCoreOptions,
+  InteractiveResultCoreProps,
+} from '../../core/interactive-result/headless-core-interactive-result';
+
+export interface InteractiveResultOptions extends InteractiveResultCoreOptions {
+  /**
+   * The query result.
+   */
+  result: Result;
+}
+
+export interface InteractiveResultProps extends InteractiveResultCoreProps {
+  /**
+   * The options for the `InteractiveResult` controller.
+   * */
+  options: InteractiveResultOptions;
+}
+
+/**
+ * The `InteractiveResult` controller provides an interface for triggering desirable side effects, such as logging UA events to the Coveo Platform, when a user selects a query result.
+ */
+export interface InteractiveResult extends InteractiveResultCore {}
+
+/**
+ * Creates an `InteractiveResult` controller instance.
+ *
+ * @param engine - The headless engine.
+ * @param props - The configurable `InteractiveResult` properties.
+ * @returns An `InteractiveResult` controller instance.
+ */
+export function buildInteractiveResult(
+  engine: ProductListingEngine,
+  props: InteractiveResultProps
+): InteractiveResult {
+  let wasOpened = false;
+
+  const logAnalyticsIfNeverOpened = () => {
+    if (wasOpened) {
+      return;
+    }
+    wasOpened = true;
+    engine.dispatch(logDocumentOpen(props.options.result));
```

Comment:
What's important is the analytics getting logged for product listing, when a document is clicked.

If everything is good on that front, 🚢

Reply:
This is a great point, the `logDocumentOpen` is a good action to use. In the case of commerce though, ML expects ~5 commerce fields in the `customData` object listed in [this sheet](https://docs.google.com/spreadsheets/d/1eegdQ4TYTYjfDmgTiqHdX97aGifCC2PhgG8kDYDJFfA/edit#gid=846427451) (see the [UA] product click tab).

The `logDocumentOpen` action today does not send `id`, `brand` etc. I think this is _okay_ for this PR because professional services are aware of the limitation, and are using middleware to enrich events with additional fields in the case of `search`. Doing the same in product listing would be consistent ...

In a follow-up PR, we could improve the `logDocumentOpen` action by adding the relevant commerce standard fields to both search and PLP clicks if/when we find them in the result. Notes:
- This is a little hacky so open to other suggestions. A commerce click action could be ideal, but requires initing the search engine with `commerce` option? Or provide a `buildCommerceSearchEngine` ...
- The solution only works if the catalog is using the standard fields. This should be a fair assumption since other functionality depends on it, but if it's not the case, middleware would still be needed.
- Ideally, only a product id (sku) would be enough, and the other fields could be enriched server-side. This would simplify the implementation for direct api users.

tl;dr I think this is good for now, and we can improve on it in the future.